beacon-chain/rpc/prysm/v1alpha1/validator/ptc_attester.go

```diff
@@ -2,16 +2,45 @@ package validator
 
 import (
 	"context"
+	"slices"
 
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/helpers"
 	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 func (vs *Server) GetPayloadAttestationData(ctx context.Context, req *ethpb.GetPayloadAttestationDataRequest) (*ethpb.PayloadAttestationData, error) {
 	return nil, errors.New("not implemented")
 }
 
-func (vs *Server) SubmitPayloadAttestation(ctx context.Context, in *ethpb.PayloadAttestationMessage) (*empty.Empty, error) {
-	return nil, errors.New("not implemented")
+// SubmitPayloadAttestation broadcasts a payload attestation message to the network and saves the payload attestation to the cache.
+// This handler does not validate the payload attestation message before broadcasting and saving it to the cache.
+// The caller should be responsible for validating the message, as it assumes a trusted relationship between the caller and the server.
+func (vs *Server) SubmitPayloadAttestation(ctx context.Context, a *ethpb.PayloadAttestationMessage) (*empty.Empty, error) {
```

Comment:
I think the right approach here is to just call the blockchain or sync handler for payload attestation messages. Instead of repeating code here. The blockchain handler is not ready yet but the sync handler is. I haven't added the blockchain handler yet because it's not clear if it'll be more than just forkchoice, if it's only forkchoice adding then we may as well add the fork choice call to the subscriber and export it.

Reply:
I think that makes sense for adding payload attestation to the cache, but I'm not sure how you would broadcast the payload attestation using the subscribe code, you might need some unusual plumbing.